packages/authorizer/test/AuthorizedHelpers.test.ts

```diff
@@ -0,0 +1,66 @@
+import { bn, deploy, ONES_ADDRESS } from '@mimic-fi/v3-helpers'
+import { expect } from 'chai'
+import { Contract } from 'ethers'
+
+/* eslint-disable no-secrets/no-secrets */
+
+describe('AuthorizedHelpers', () => {
+  let authorizedHelpers: Contract
+
+  before('create authorizer', async () => {
+    authorizedHelpers = await deploy('AuthorizedHelpersMock', [])
+  })
+
+  describe('authParams', () => {
+    const address = ONES_ADDRESS
+    const bytes32 = '0x0000000000000000000000000000000000000000000000000000000000000002'
+    const uint256 = bn(100)
+    const bool = true
+    const bytes4 = '0x00000001'
+
+    function itBehavesLikeAuthParams(params: string, ...args) {
+      it('creates the array properly', async () => {
+        const functionSignature = 'getAuthParams(' + params + ')'
+        const result = await authorizedHelpers[functionSignature](...args)
+
+        expect(result.length).to.be.equal(args.length)
+
+        result.forEach((v, i) => {
+          const arg = args[i]
+          const expectedValue = typeof arg === 'boolean' ? Number(arg) : arg
+          expect(v).to.be.equal(expectedValue)
+        })
+      })
+    }
```

Comment:
WDYT if we do the following?

```suggestion
    function itBehavesLikeAuthParams(paramsSig: string) {
      it('creates the array properly', async () => {
        const args = paramsSig.split(',').map((t, i) => {
          if (t == 'boolean') return true
          if (t == 'bytes4') return utils.hexZeroPad(i + 1, 4)
          if (t == 'bytes32') return utils.hexZeroPad(i + 1, 32)
          if (t == 'address') return utils.hexZeroPad(i + 1, 20)
          return i+1
        })

        const result = await authorizedHelpers[`getAuthParams(${paramsSig})`](...args)
        expect(result.length).to.be.equal(args.length)

        result.forEach((actualValue, i) => {
          const arg = args[i]
          const expectedValue = typeof arg === 'boolean' ? Number(arg) : arg
          expect(actualValue).to.be.equal(expectedValue)
        })
      })
    }
```

Otherwise, testing an scenario like `itBehavesLike('uint256,uint256', 1, 2)` could trigger a false positive